embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/telemetry/EmbraceTelemetryService.kt

```diff
@@ -2,76 +2,59 @@ package io.embrace.android.embracesdk.telemetry
 
 import io.embrace.android.embracesdk.capture.metadata.EmbraceMetadataService
 import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
-import java.util.Locale
+import io.embrace.android.embracesdk.internal.spans.toEmbraceUsageAttributeName
+import java.util.concurrent.ConcurrentHashMap
 
-/*
-    Service for tracking usage of public APIs, and different internal metrics about the app.
+/**
+ * Service for tracking usage of public APIs, and different internal metrics about the app.
  */
 internal class EmbraceTelemetryService(
     private val okHttpReflectionFacade: OkHttpReflectionFacade = OkHttpReflectionFacade()
-) {
+) : TelemetryService {
 
-    private val usageCountMap = mutableMapOf<String, Int>()
-    private val appAttributesMap = mutableMapOf<String, String>()
+    private val usageCountMap = ConcurrentHashMap<String, Int>()
+    private val appAttributes: Map<String, String> by lazy { computeAppAttributes() }
 
-    /*
-        Tracks the usage of a public API by name. Adds a suffix for easier identification. Replaces whitespaces with underscores.
-    */
-    fun onPublicApiCalled(name: String) {
-        val suffixedName = "emb.usage.${
-        name.toLowerCase(Locale.ENGLISH)
-            .trim()
-            .replace(" ", "_")
-        }"
-        usageCountMap[suffixedName] = (usageCountMap[suffixedName] ?: 0) + 1
+    override fun onPublicApiCalled(name: String) {
+        usageCountMap[name] = (usageCountMap[name] ?: 0) + 1
     }
 
-    /*
-        Returns a map with every telemetry value. For now, it's just usage counts of public APIs.
-     */
-    fun getTelemetryAttributes(): Map<String, String> {
-        val telemetryMap = mutableMapOf<String, String>()
-
-        telemetryMap.putAll(getUsageCountTelemetry())
-
-        telemetryMap.putAll(getAppAttributes())
-
-        return telemetryMap
+    override fun getTelemetryAttributes(): Map<String, String> {
```

Comment:
Actually, it's the public method that should make it clear that the getter is destructive. We don't want consumers of this API to call this multiple times because they are unaware that i'll clear the underlying counts.